src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -218,43 +403,58 @@ void extractRecurrence(const QJsonArray &recurrence, KCalCore::Event::Ptr event,
         return;
     }
 
+    kcalRecurrence->clear(); // avoid adding duplicate recurrence information
```

Comment:
This single line resolves the issue where after a sync, 2 RRULE sections can exist within a single recurrence() for an event (as reported by event->recurrence()->dump() from within convertRecurrence() of calendarworker.cpp from nemo-qml-plugin-calendar, with KCALDEBUG=1 set, when running jolla-calendar).